### ISSUE TITLE
Most integer types silently ignored by identify

### ISSUE THREAD

**Issue by jeiting** [Appboy/appboy-segment-ios#1]
```
for (NSString *key in payload.traits.allKeys) {
    if (![appboyTraits containsObject:key]) {
      id traitValue = payload.traits[key];
      if ([traitValue isKindOfClass:[NSString class]]) {
      [[Appboy sharedInstance].user setCustomAttributeWithKey:key andStringValue:traitValue];
        SEGLog(@"[[Appboy sharedInstance].user setCustomAttributeWithKey: andStringValue:]");
      } else if ([traitValue isKindOfClass:[NSNumber class]]) {
        if (strcmp([traitValue objCType], [@(YES) objCType]) == 0) {
          [[Appboy sharedInstance].user setCustomAttributeWithKey:key andBOOLValue:[(NSNumber *)traitValue boolValue]];
          SEGLog(@"[[Appboy sharedInstance].user setCustomAttributeWithKey: andBOOLValue:]");
        } else if (strcmp([traitValue objCType], @encode(int)) == 0) {
          [[Appboy sharedInstance].user setCustomAttributeWithKey:key andIntegerValue:[(NSNumber *)traitValue integerValue]];
          SEGLog(@"[[Appboy sharedInstance].user setCustomAttributeWithKey: andIntegerValue:]");
        } else if (strcmp([traitValue objCType], @encode(double)) == 0) {
          [[Appboy sharedInstance].user setCustomAttributeWithKey:key andDoubleValue:[(NSNumber *)traitValue doubleValue]];
          SEGLog(@"[[Appboy sharedInstance].user setCustomAttributeWithKey: andDoubleValue:]");
        } else {
          SEGLog(@"Could not map NSNumber value to Appboy custom attribute:%@]", traitValue);
        }
      }
    }
  }
```

This code will not be able to map any integer type that isn't cast to an `int` type, this includes `long`s and `NSIntegers`, basically most of the integer types that are used in iOS programming.

**Comment by briancaw** [comment]
Hi @jeiting 

Thanks for reporting this.  We're looking into this and should have a fix out in the near future.  We'll update with a timeline ASAP.

Thanks,
Brian

**Comment by jeiting** [comment]
This bug also extends to `float`s

**Comment by Wenzhi** [comment]
Hi @jeiting,

Our iOS SDK excepts `double` for custom attributes, instead of `float`. You can see the code [here](https://github.com/Appboy/appboy-ios-sdk/blob/master/AppboyKit/headers/AppboyKitLibrary/ABKUser.h#L198). 

Is there any reason that you cannot use `double` but must use `float` for custom attributes? We’re happy to consider modifying our interface given more information

Thanks,
Wenzhi

**Comment by jeiting** [comment]
This will continue to be a point of confusion because generally passing a float in place of double will be handled, the linked example will automatically upcast to a double. For the segment integration it just fails silently and does nothing. This means that any time I pass an `NSNumber` I have to make sure its internal representation matches one that the SEG Appboy integration can handle.

**Comment by Wenzhi** [comment]
Hi @jeiting,

We just released a new SDK 1.0.2 which also supports short, float and long types as custom attribute values. If you could test and confirm it resolves your issue it'd be greatly appreciated.

Thanks,
Wenzhi

**Comment by briancaw** [comment]
Hi @jeiting 

Thanks again for reporting this.  Please re-open if you find any part of our solution can/should be improved.

Thanks,
Brian
